javagen/src/main/java/com/azure/autorest/model/clientmodel/ClientMethod.java

```diff
@@ -320,11 +320,16 @@ public void addImportsTo(Set<String> imports, boolean includeImplementationImpor
         if (settings.isDataPlaneClient()) {
             imports.add("com.azure.core.http.HttpMethod");
             imports.add("com.azure.core.http.rest.Response");
-            imports.add("com.azure.core.http.rest.RequestOptions");
-            imports.add("com.azure.core.util.BinaryData");
-            imports.add(DateTimeRfc1123.class.getName());
+            imports.add(ClassType.RequestOptions.getFullName());
+            imports.add(ClassType.BinaryData.getFullName());
+            imports.add(ClassType.DateTimeRfc1123.getFullName());
+
+            for (ClientMethodParameter parameter : getParameters()) {
+                parameter.addImportsTo(imports, includeImplementationImports);
+            }
 
             if (includeImplementationImports) {
+                imports.add(ClassType.Context.getFullName());
                 imports.add("com.azure.core.http.rest.SimpleResponse");
                 imports.add("com.azure.core.http.HttpRequest");
                 if (settings.getAddContextParameter() || settings.isContextClientMethodParameter()) {
```

Comment:
We should refactor this code.

There is little reason to completely branch on dpg or not.